src/renderer/hooks/useProcessQr.ts

```diff
@@ -47,43 +67,110 @@ export default function useProcessQR() {
     [openAlertDialog]
   )
 
+  // Users can enter the "Instant Onboarding" flow by scanning a DCACCOUNT,
+  // DC_ASK_VERIFYGROUP, or DC_ASK_VERIFYCONTACT code which essentially creates
+  // a new "chatmail" profile for them and connects them with the regarding
+  // contact or group if given.
+  //
+  // We ask the user if they really want to proceed with this action and
+  // accordingly prepare the onboarding.
+  const startInstantOnboarding = useCallback(
+    async (accountId: number, qrWithUrl: QrWithUrl) => {
+      const { qr } = qrWithUrl
+      const numberOfAccounts = (await BackendRemote.rpc.getAllAccountIds())
+        .length
+
+      if (qr.kind === 'account' && numberOfAccounts > 0) {
+        // Ask user to confirm creating a new account if they already have one
+        const message: string =
+          numberOfAccounts === 1
+            ? 'qraccount_ask_create_and_login'
+            : 'qraccount_ask_create_and_login_another'
+        const userConfirmed = await openConfirmationDialog({
+          message: tx(message, qr.domain),
+          confirmLabel: tx('login_title'),
+        })
+
+        if (!userConfirmed) {
+          return
+        }
+      } else if (qr.kind === 'askVerifyGroup') {
+        // Ask the user if they want to create a new account and join the group
+        const userConfirmed = await openConfirmationDialog({
+          message: tx('instant_onboarding_confirm_group', qr.grpname),
+          confirmLabel: tx('ok'),
+        })
+
+        if (!userConfirmed) {
+          return
+        }
+      } else if (qr.kind === 'askVerifyContact') {
+        // Ask the user if they want to create a new account and start
+        // chatting with contact
+        const contact = await BackendRemote.rpc.getContact(
+          accountId,
+          qr.contact_id
+        )
+
+        const userConfirmed = await openConfirmationDialog({
+          message: tx('instant_onboarding_confirm_contact', contact.address),
+          confirmLabel: tx('ok'),
+        })
+
+        if (!userConfirmed) {
+          return
+        }
+      }
+
+      // Start instant onboarding flow for user
+      await switchToInstantOnboarding(qrWithUrl)
```

Comment:
naming confusion: why not call it like the comment: `startInstantOnboardingFlow`.

Also the comment "for user" is a bit confusing as below a new account could be created, so it's not clear who this "user" is.

Reply:
Similar to my other comment, not sure if I can follow, feel free to do changes directly